src/shared/pipes/self-or-empty.pipe.spec.ts

```diff
@@ -0,0 +1,79 @@
+/*
+ * This file is part of ng2-json-editor.
+ * Copyright (C) 2016 CERN.
+ *
+ * ng2-json-editor is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * ng2-json-editor is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with ng2-json-editor; if not, write to the Free Software Foundation, Inc.,
+ * 59 Temple Place, Suite 330, Boston, MA 02111-1307, USA.
+ * In applying this license, CERN does not
+ * waive the privileges and immunities granted to it by virtue of its status
+ * as an Intergovernmental Organization or submit itself to any jurisdiction.
+ */
+
+import { EmptyValueService } from '../services/empty-value.service';
+import { SelfOrEmptyPipe } from './self-or-empty.pipe';
+import {fromJS, List} from 'immutable';
+
+describe('SelfOrEmptyPipe', () => {
+    let pipe: SelfOrEmptyPipe;
+
+    beforeEach(() => {
+        pipe = new SelfOrEmptyPipe(new EmptyValueService());
+    });
+
+    it('should return empty value if given value is undefined', () => {
+        let schema = {
+            type: 'object',
+            properties: {
+                prop1: {
+                    type: 'string'
+                }
+            }
+        };
+
+        let value = undefined;
+        let pipeResult = pipe.transform(value, schema);
+        expect(pipeResult).toBeDefined();
+    });
+
+    it('should return empty value if given value is immutable List and is empty', () => {
+        let schema = {
+            type: 'array',
+            items: {
+                type: 'string'
+            }
+        };
+
+        let value: List = fromJS([]);
```

Comment:
for creating immutables with simple javascript object please use constructors instead of `fromJS` 
Example: `let value = List<WhateverType>();`

After this update it will be fixed but this has actually syntax error, you can use `List` without giving element type.